### ISSUE TITLE
source mapping to enhance the debugging process

### ISSUE THREAD

**Issue by princesust** [abhijithvijayan/web-extension-starter#29]
Is it possible to show the actual file contents inside source map without including webpack import. which sometimes difficult to debug 

```javascript
webextension_polyfill__WEBPACK_IMPORTED_MODULE_2___default.a.runtime.onMessage.addListener(function (msg) {
  var action = msg.action;
  console.log("msg::", msg);

  if (action == _message_actions__WEBPACK_IMPORTED_MODULE_7__["default"].IS_SCRIPT_INJECTED) {
    return Promise.resolve(true);
  }
});
```

**Comment by abhijithvijayan** [comment]
Let me look into that.

**Comment by abhijithvijayan** [comment]
I am not sure if this is even possible, but I will try to change webpack configuration, hoping to resolve this.

**Comment by slhck** [comment]
Any updates on this? This would be really crucial to have. Otherwise, awesome project!

**Comment by abhijithvijayan** [comment]
Sorry, I was engaged in some projects.

I will try to find if this is possible asap.

**Comment by slhck** [comment]
I found that `devtool: "source-map",` in the WebPack config at least generates the source maps for the browser polyfill, however it creates the following issue:

```
DevTools failed to parse SourceMap: chrome-extension://biipfmkokfahmjfnnbpegccidgoldhom/js/background.bundle.js.map
```

**Comment by abhijithvijayan** [comment]
The option `devtool: 'source-map', should work.

I will see what the error is.

**Comment by abhijithvijayan** [comment]
Also, on a quick look, the solution you suggested should work.

Could you add that line at top and submit a PR?

**Comment by slhck** [comment]
I am not sure this works. I tried the solution suggested in the SO post to no avail.

Looks as follows in Chrome:

![image](https://user-images.githubusercontent.com/582444/79552490-6326e100-809b-11ea-8789-cdb60e5ef448.png)

And here in Firefox: 

![image](https://user-images.githubusercontent.com/582444/79552648-95d0d980-809b-11ea-9240-94fddf603d1f.png)

Basically I think that the hot reloading module is interfering here.

**Comment by abhijithvijayan** [comment]
I think it is the issue with `webextension-polyfill`

There is a similar issue [here](https://github.com/AliasIO/wappalyzer/issues/2932)

**Comment by abhijithvijayan** [comment]
Could you see if everything works with the devtool option enabled and the reloader disabled?

**Comment by slhck** [comment]
I tested that, and I get the same error. The source code shown is basically the same bundled code, except for the reloader message on top. So the issue is a different one, and probably has to do with webpack itself not providing the source map correctly.

**Comment by abhijithvijayan** [comment]
@slhck @princesust I guess it's an upstream issue. https://github.com/abhijithvijayan/web-extension-starter/issues/29

**Comment by abhijithvijayan** [comment]
@princesust @slhck @antons

See if the fix 23cc204 works

Closing for now.

Feel free to re-open or comment if the issue still persist.

Thanks for reporting the issue

**Comment by slhck** [comment]
This seems to work! Thank you! :tada:
